src/lib.rs

```diff
@@ -82,16 +81,14 @@ impl vm::Ext for EwasmExt {
     fn origin_balance(&self) -> Result<U256> {
         // NOTE: used by SLEFDESTRUCT for gas metering (not used here now since we don't charge gas)
         let origin = ewasm_api::tx_origin();
-        Ok(U256::from(U128::from_little_endian(
-            &ewasm_api::external_balance(&origin).bytes,
-        )))
+        let balance: [u8; 16] = ewasm_api::external_balance(&origin).into();
```

Comment:
Is this part not messing with endianess?

Reply:
Good catch. There were 3 more instances of the same issue, fixed them as well